[distributed] add test to ensure that dist autograd contexts are cleaned up in
the case of nested rpcs

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#28485 [distributed] add test to ensure that dist autograd contexts are cleaned up in
the case of nested rpcs**

Closes #28124. This diff adds a test to ensure that when we have multiple nested RPCs
inside a dist autograd context, the context that is created as a result of a
nested rpc is cleaned up after the node creating the context exits the context
manager. For example, worker 0 might send an rpc to worker 1 that results in an
rpc to worker 2, so worker 2 will have 0's context, even though worker 0 never
directly talked to 2. This test ensures that the context on 2 would also be
cleaned up.

Differential Revision: [D18079212](https://our.internmc.facebook.com/intern/diff/D18079212/)

rohan-varma committed:
[distributed] add test to ensure that dist autograd contexts are cleaned up in
the case of nested rpcs

This diff adds a test to ensure that when we have multiple nested RPCs
inside a dist autograd context, the context that is created as a result of a
nested rpc is cleaned up after the node creating the context exits the context
manager. For example, worker 0 might send an rpc to worker 1 that results in an
rpc to worker 2, so worker 2 will have 0's context, even though worker 0 never
directly talked to 2. This test ensures that the context on 2 would also be
cleaned up.

Differential Revision: [D18079212](https://our.internmc.facebook.com/intern/diff/D18079212/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[distributed] add test to ensure that dist autograd contexts are cleaned up in
the case of nested rpcs"


This diff adds a test to ensure that when we have multiple nested RPCs
inside a dist autograd context, the context that is created as a result of a
nested rpc is cleaned up after the node creating the context exits the context
manager. For example, worker 0 might send an rpc to worker 1 that results in an
rpc to worker 2, so worker 2 will have 0's context, even though worker 0 never
directly talked to 2. This test ensures that the context on 2 would also be
cleaned up.

Differential Revision: [D18079212](https://our.internmc.facebook.com/intern/diff/D18079212/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[distributed] add test to ensure that dist autograd contexts are cleaned up in
the case of nested rpcs"

the case of nested rpcs**

This diff adds a test to ensure that when we have multiple nested RPCs
inside a dist autograd context, the context that is created as a result of a
nested rpc is cleaned up after the node creating the context exits the context
manager. For example, worker 0 might send an rpc to worker 1 that results in an
rpc to worker 2, so worker 2 will have 0's context, even though worker 0 never
directly talked to 2. This test ensures that the context on 2 would also be
cleaned up.

Differential Revision: [D18079212](https://our.internmc.facebook.com/intern/diff/D18079212/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[distributed] add test to ensure that dist autograd contexts are cleaned up in
the case of nested rpcs"

the case of nested rpcs**

Closes #28124. This diff adds a test to ensure that when we have multiple nested RPCs
inside a dist autograd context, the context that is created as a result of a
nested rpc is cleaned up after the node creating the context exits the context
manager. For example, worker 0 might send an rpc to worker 1 that results in an
rpc to worker 2, so worker 2 will have 0's context, even though worker 0 never
directly talked to 2. This test ensures that the context on 2 would also be
cleaned up.

Differential Revision: [D18079212](https://our.internmc.facebook.com/intern/diff/D18079212/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[distributed] add test to ensure that dist autograd contexts are cleaned up in
the case of nested rpcs"

the case of nested rpcs**
the case of nested rpcs**

Closes #28124. This diff adds a test to ensure that when we have multiple nested RPCs
inside a dist autograd context, the context that is created as a result of a
nested rpc is cleaned up after the node creating the context exits the context
manager. For example, worker 0 might send an rpc to worker 1 that results in an
rpc to worker 2, so worker 2 will have 0's context, even though worker 0 never
directly talked to 2. This test ensures that the context on 2 would also be
cleaned up.

Differential Revision: [D18079212](https://our.internmc.facebook.com/intern/diff/D18079212/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[distributed] add test to ensure that dist autograd contexts are cleaned up in
the case of nested rpcs"



Closes #28124. This diff adds a test to ensure that when we have multiple nested RPCs
inside a dist autograd context, the context that is created as a result of a
nested rpc is cleaned up after the node creating the context exits the context
manager. For example, worker 0 might send an rpc to worker 1 that results in an
rpc to worker 2, so worker 2 will have 0's context, even though worker 0 never
directly talked to 2. This test ensures that the context on 2 would also be
cleaned up.

Differential Revision: [D18079212](https://our.internmc.facebook.com/intern/diff/D18079212/)

[ghstack-poisoned]

zhaojuanmao reviewed (approved):
overall look good, although feel like we can merge _set_rpc_done and _store_context_id and clean them up

rohan-varma committed:
Update on "[distributed] add test to ensure that dist autograd contexts are cleaned up in
the case of nested rpcs"

the case of nested rpcs**


Closes #28124. This diff adds a test to ensure that when we have multiple nested RPCs
inside a dist autograd context, the context that is created as a result of a
nested rpc is cleaned up after the node creating the context exits the context
manager. For example, worker 0 might send an rpc to worker 1 that results in an
rpc to worker 2, so worker 2 will have 0's context, even though worker 0 never
directly talked to 2. This test ensures that the context on 2 would also be
cleaned up.

Differential Revision: [D18079212](https://our.internmc.facebook.com/intern/diff/D18079212/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[distributed] add test to ensure that dist autograd contexts are cleaned up in
the case of nested rpcs"

the case of nested rpcs**
the case of nested rpcs**


Closes #28124. This diff adds a test to ensure that when we have multiple nested RPCs
inside a dist autograd context, the context that is created as a result of a
nested rpc is cleaned up after the node creating the context exits the context
manager. For example, worker 0 might send an rpc to worker 1 that results in an
rpc to worker 2, so worker 2 will have 0's context, even though worker 0 never
directly talked to 2. This test ensures that the context on 2 would also be
cleaned up.

Differential Revision: [D18079212](https://our.internmc.facebook.com/intern/diff/D18079212/)

[ghstack-poisoned]

rohan-varma commented:
@zhaojuanmao updated the PR to remove `_store_context_id` as suggested

rohan-varma committed:
Update on "[distributed] add test to ensure that dist autograd contexts are cleaned up in
the case of nested rpcs"


Closes #28124. This diff adds a test to ensure that when we have multiple nested RPCs
inside a dist autograd context, the context that is created as a result of a
nested rpc is cleaned up after the node creating the context exits the context
manager. For example, worker 0 might send an rpc to worker 1 that results in an
rpc to worker 2, so worker 2 will have 0's context, even though worker 0 never
directly talked to 2. This test ensures that the context on 2 would also be
cleaned up.

Differential Revision: [D18079212](https://our.internmc.facebook.com/intern/diff/D18079212/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[distributed] add test to ensure that dist autograd contexts are cleaned up in
the case of nested rpcs"

the case of nested rpcs**

Closes #28124. This diff adds a test to ensure that when we have multiple nested RPCs
inside a dist autograd context, the context that is created as a result of a
nested rpc is cleaned up after the node creating the context exits the context
manager. For example, worker 0 might send an rpc to worker 1 that results in an
rpc to worker 2, so worker 2 will have 0's context, even though worker 0 never
directly talked to 2. This test ensures that the context on 2 would also be
cleaned up.

Differential Revision: [D18079212](https://our.internmc.facebook.com/intern/diff/D18079212/)

[ghstack-poisoned]

zhaojuanmao commented:
sounds good!
